### DISCUSSION TITLE
How to detect upcoming turn directions - left/ right ?

### DISCUSSION THREAD

**Discussion by shatroopa-si** [carla-simulator/carla#7253] [Q&A]
Hi,
I am trying to detect direction of upcoming turn from the simulator using _waypoint.lane_change along with checks for intersection and junction_ attributes but it is not giving the right output. Sometimes, when there's a barrier between 2 lanes and a turn is on the other side of the barrier, it still shows the turn. On other times, it misses a valid turn (A turn that can be taken as per traffic rules). It has also reported a left turn to be a right turn and vice versa on some instances.
Carla version: 0.9.15

**Comment by Daraan** [comment]
Can you point to a concrete example, i.e. map & location?

Might sharing your code?

---

I likely will look into this myself in the next weeks.
